tls/s2n_server_extensions.c

```diff
@@ -191,3 +203,14 @@ int s2n_recv_server_sct_list(struct s2n_connection *conn, struct s2n_stuffer *ex
 
     return 0;
 }
+
+int s2n_recv_server_max_frag_len(struct s2n_connection *conn, struct s2n_stuffer *extension)
+{
+    uint16_t max_fragment_length;
+    GUARD(s2n_stuffer_read_uint16(extension, &max_fragment_length));
+    if (max_fragment_length != mfl_code_to_length[conn->config->mfl_code]) {
+        S2N_ERROR(S2N_ERR_MAX_FRAG_LEN_MISMATCH);
```

Comment:
Good catch: According to RFC, I say we should make it a hard error and terminate the handshake. What do you think?

RFC6066: " if a client receives a maximum fragment length negotiation response that differs from the length it requested, it MUST also abort the handshake with an "illegal_parameter" alert."